persistence/sql/migrations/20211019000001000000_merge_authentication_request_tables.sqlite.up.sql

```diff
@@ -0,0 +1,187 @@
+CREATE INDEX hydra_oauth2_flow_login_session_id_idx ON hydra_oauth2_flow (login_session_id);
+CREATE INDEX hydra_oauth2_flow_subject_idx ON hydra_oauth2_flow (subject);
+CREATE UNIQUE INDEX hydra_oauth2_flow_verifier_idx ON hydra_oauth2_flow (verifier);
+
+
+-- Recreate hydra_oauth2_consent_request in order to rename a foreign key reference
+
+CREATE TABLE hydra_oauth2_consent_request_tmp
+(
+    challenge                 VARCHAR(40)  NOT NULL PRIMARY KEY,
+    verifier                  VARCHAR(40)  NOT NULL,
+    client_id                 VARCHAR(255) NOT NULL REFERENCES hydra_client (id) ON DELETE CASCADE,
+    subject                   VARCHAR(255) NOT NULL,
+    request_url               TEXT         NOT NULL,
+    skip                      INTEGER      NOT NULL,
+    requested_scope           TEXT         NOT NULL,
+    csrf                      VARCHAR(40)  NOT NULL,
+    authenticated_at          TIMESTAMP    NULL,
+    requested_at              TIMESTAMP    NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    oidc_context              TEXT         NOT NULL,
+    forced_subject_identifier VARCHAR(255) NULL     DEFAULT '',
+    login_session_id          VARCHAR(40)  NULL REFERENCES hydra_oauth2_authentication_session (id) ON DELETE SET NULL,
+    login_challenge           VARCHAR(40)  NULL REFERENCES hydra_oauth2_flow (challenge) ON DELETE SET NULL,
+    requested_at_audience     TEXT         NULL     DEFAULT '',
+    acr                       TEXT         NULL     DEFAULT '',
+    amr                       TEXT         NOT NULL DEFAULT '',
+    context                   TEXT         NOT NULL DEFAULT '{}'
+);
+
+INSERT INTO hydra_oauth2_consent_request_tmp (
+     challenge,
+     verifier,
+     client_id,
+     subject,
+     request_url,
+     skip,
+     requested_scope,
+     csrf,
+     authenticated_at,
+     requested_at,
+     oidc_context,
+     forced_subject_identifier,
+     login_session_id,
+     login_challenge,
+     requested_at_audience,
+     acr,
+     amr,
+     context
+) SELECT
+     challenge,
+     verifier,
+     client_id,
+     subject,
+     request_url,
+     skip,
+     requested_scope,
+     csrf,
+     authenticated_at,
+     requested_at,
+     oidc_context,
+     forced_subject_identifier,
+     login_session_id,
+     login_challenge,
+     requested_at_audience,
+     acr,
+     amr,
+     context
+FROM hydra_oauth2_consent_request;
+
+DROP TABLE hydra_oauth2_consent_request;
+ALTER TABLE hydra_oauth2_consent_request_tmp RENAME TO hydra_oauth2_consent_request;
+
+CREATE INDEX hydra_oauth2_consent_request_client_id_idx ON hydra_oauth2_consent_request (client_id);
+CREATE INDEX hydra_oauth2_consent_request_subject_idx ON hydra_oauth2_consent_request (subject);
+CREATE INDEX hydra_oauth2_consent_request_login_session_id_idx ON hydra_oauth2_consent_request (login_session_id);
+CREATE INDEX hydra_oauth2_consent_request_login_challenge_idx ON hydra_oauth2_consent_request (login_challenge);
```

Comment:
Are we missing any keys here and if not where can we find the reference to existing keys at the time of this migration?

Reply:
The way I make these sqlite migrations is that I dump the current schema, copy the relevant parts, and make the necessary changes. Maybe we could automatically create schema diffs for each migration and version these diffs to make the migrations better reviewable?

Reply:
Here's a bash oneliner to diff the sqlite schema of two commits: `__dump() { P="/tmp/sqlitest/$1.sqlite"; rm $P; yes | go run -tags sqlite . migrate sql "sqlite://$P?_fk=true"; echo '.dump' | sqlite3 $P > /tmp/sqlitest/$1.dump; }; __diff_sqlite() { mkdir -p /tmp/sqlitest; git checkout $1; __dump $1; git checkout $2; __dump $2; git diff --no-index /tmp/sqlitest/$1.dump /tmp/sqlitest/$2.dump; }`. Usage: `__diff_sqlite c0f38431 08a1aa35`

Reply:
Ok, I see! That is a good approach :) Maybe add some docs in the PR on the final commit so we can ensure the SQL schemas do indeed match as much as possible?

Reply:
Will do :)